source/MaterialXCore/Interface.cpp

```diff
@@ -317,10 +327,17 @@ InputPtr InterfaceElement::getActiveInput(const string& name) const
 vector<InputPtr> InterfaceElement::getActiveInputs() const
 {
     vector<InputPtr> activeInputs;
+    ValueElementSet activeInputsSet(ValueElementLess);
```

Comment:
For outputs, inputs, and value elements in general, to get the activeOutputs, activeInputs, and activeValueElements, use a set to avoid duplicating inherited value elements. This was for example a parent NodeDef can have an output called "out" and the child NodeDef can have the same output called "out" and we wont end up with two outputs, but just one. This is also particularly useful for inputs as the input that is most derived will be used over the parent input allowing it to be more specialized.